### PR TITLE
Refactor grading service V11 tests

### PR DESCRIPTION
**Pull request by jon-betts**
This is a grab bag of various improvements. Each has been put in it's own commit for easy reference.

Highlights:

 * Using sentinels
 * Making test which apply to both methods use both methods
 * Collapsing some tests together
 * More declarative comparisons with dicts

### CODE REVIEW

File: tests/unit/lms/services/lti_grading/_v11_test.py [hunk 1]
```diff
@@ -9,88 +9,162 @@
...
+                    },
+                    "imsx_POXBody": {
+                        "readResultResponse": {
+                            "result": {
+                                "resultScore": {"language": "en", "textString": score}
+                            }
+                        }
+                    },
+                }
+            }
```
Comment (jon-betts) [inline tests/unit/lms/services/lti_grading/_v11_test.py:42]: I think this much more clearly shows the data structure at play than before. The strategy is we will create the full thing, and then knock bits out or modify them as required for the tests.

I've left in non-conditional things like score and status code as it's easier to do this way.

File: tests/unit/lms/services/lti_grading/_v11_test.py [hunk 2]
```diff
@@ -135,39 +211,7 @@ def test_requests_fail_if_body_not_xml(self, svc, http_service):
...
-            match="<imsx_description>An error occurred.</imsx_description>",
-        ):
-            svc.read_result(self.GRADING_ID)
-
-    def test_requests_fail_and_no_description_returned(self, svc, respond_with):
-        respond_with(status_code="failure", include_description=False)
-
-        # imsx_description is missing
-        with pytest.raises(ExternalRequestError, match="LTI outcome request failed"):
-            svc.read_result(self.GRADING_ID)
```
Comment (jon-betts) [inline tests/unit/lms/services/lti_grading/_v11_test.py:170]: All of these are actually `read_result` tests, but for some reason aren't named like that. They also are positioned after the tests for the other method. I've renamed these and put them to the top.

File: tests/unit/lms/services/lti_grading/_v11_test.py [hunk 3]
```diff
@@ -178,100 +222,24 @@ def sent_pox_body(cls, http_service):
...
     @classmethod
-    def assert_sent_header_ok(cls, http_service):
-        """Check standard header fields of the request body."""
-
-        body = cls.sent_body(http_service)
-
-        header = body["imsx_POXEnvelopeRequest"]["imsx_POXHeader"]
-        message_id = header["imsx_POXRequestHeaderInfo"]["imsx_messageIdentifier"]
-
-        assert message_id == "999999123"
```
Comment (jon-betts) [inline tests/unit/lms/services/lti_grading/_v11_test.py:189]: This is being done in tests now

File: tests/unit/lms/services/lti_grading/_v11_test.py [hunk 4]
```diff
@@ -178,100 +222,24 @@ def sent_pox_body(cls, http_service):
...
-            {
-                "imsx_POXEnvelopeResponse": {
-                    "@xmlns": "http://www.imsglobal.org/services/ltiv1p1/xsd/imsoms_v1p0",
-                    "imsx_POXHeader": {"imsx_POXResponseHeaderInfo": header_info},
-                    "imsx_POXBody": {
-                        "readResultResponse": {"result": {"resultScore": result_score}}
-                    },
-                }
-            }
-        )
```
Comment (jon-betts) [inline tests/unit/lms/services/lti_grading/_v11_test.py:223]: This is part of the conditionality which is removed to the top

File: tests/unit/lms/services/lti_grading/_v11_test.py [hunk 5]
```diff
@@ -178,100 +222,24 @@ def sent_pox_body(cls, http_service):
...
-            {
-                # "imsx_POXEnvelopeResponse" is the expected key. This erroneous value is inspired
-                # by Blackbaud.
-                "imsx_POXEnvelopeRequest": {
-                    "@xmlns": "http://www.imsglobal.org/services/ltiv1p1/xsd/imsoms_v1p0",
-                    "imsx_POXHeader": {},
-                    "imsx_POXBody": {},
-                }
-            }
-        )
```
Comment (jon-betts) [inline tests/unit/lms/services/lti_grading/_v11_test.py:237]: Moved into the one test that needs it and simplified.

File: tests/unit/lms/services/lti_grading/_v11_test.py [hunk 6]
```diff
@@ -178,100 +222,24 @@ def sent_pox_body(cls, http_service):
...
+        def respond_with(response, status=200):
             http_service.post.return_value = factories.requests.Response(
                 status_code=status,
-                raw=response_body,
+                raw=xmltodict.unparse(response),
                 content_type="application/xml",
             )
 
-        return configure
+        return respond_with
```
Comment (jon-betts) [inline tests/unit/lms/services/lti_grading/_v11_test.py:241]: We now accept a dict here and XML-ify it

**Review by marcospri — APPROVED** [review]

**Pull request merged**